#!/usr/bin/env python
PACKAGE = "iiwa_toolkit"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# DS Gains
gen.add("Position_DSgain",   double_t,    0, "scalling the preset ds gain from yaml", 1,  0.5, 2)
gen.add("Orientation_DSgain",   double_t,    0, "scalling the preset ds gain from yaml", 1,  0.5, 2)

# Passive Gains
gen.add("Position_lambda",   double_t,    0, "scalling the preset lambda0 and lambda1 from yaml", 1,  0.5, 2)
gen.add("Orientation_lambda",   double_t,    0, "scalling the preset lambda0 and lambda1 from yaml", 1,  0.5, 2)

# leader Positon
gen.add("Leader_dX",   double_t,    0, "displacing the preset des_position from yaml", 0,  -0.5, 0.5)
gen.add("Leader_dY",   double_t,    0, "displacing the preset des_position from yaml", 0,  -0.5, 0.5)
gen.add("Leader_dZ",   double_t,    0, "displacing the preset des_position from yaml", 0,  -0.5, 0.5)

gen.add("mirroring_x",   bool_t,   0, "A Boolean parameter",  False)
gen.add("mirroring_y",   bool_t,   0, "A Boolean parameter",  False)
gen.add("mirroring_z",   bool_t,   0, "A Boolean parameter",  False)

# Desired Orientation
gen.add("dX_des_angle",   double_t,    0, "displacing the preset des_orientation from yaml", 0,  -1.5, 1.5)
gen.add("dY_des_angle",   double_t,    0, "displacing the preset des_orientation from yaml", 0,  -1.5, 1.5)
gen.add("dZ_des_angle",   double_t,    0, "displacing the preset des_orientation from yaml", 0,  -1.5, 1.5)


exit(gen.generate(PACKAGE, "iiwa_toolkit", "marker_tracking_cfg_params"))